databases/business-sync/fragment-shoebox-snippet.js

```diff
@@ -0,0 +1,40 @@
+{
+  channels: getDocSyncChannels(doc, oldDoc),
+  typeFilter: function(doc, oldDoc) {
+    return createBusinessEntityRegex('shoeboxSnippet\\.[A-Za-z0-9_-]+\\.[A-Za-z0-9_-]+').test(doc._id);
+  },
+  allowUnknownProperties: true,
+  immutable: true,
+  propertyValidators: {
+    type: {
+      // The type of shoebox snippet (ie, bank transaction, email, photo, digital receipt)
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    source: {
+      // The source of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    },
+    kashooEntityType: {
+      // Type of corresponding Books entity
+      type: 'string',
+      required: false,
+      mustNotBeEmpty: true
+    },
+    kashooId: {
+      // The ID of the corresponding Books entity
+      type: 'integer',
+      required: false,
+      minimumValue: 1
+    },
+    data: {
+      // Raw data of the snippet
+      type: 'string',
+      required: true,
+      mustNotBeEmpty: true
+    }
```

Comment:
Definitely a +1 on a source identifier.

I can't wrap my head around `processed` yet. It does seem like we will want to have something which will allow us to easily return the ones that need to be reviewed and processed. If it is automation that is doing it then efficiency of the query isn't necessarily a concern at this point. However, if we're looking to show a shoebox listing of items that need attention we'll be looking to pull the unprocessed items regularly and quickly.

Another twist to throw into this, and it may apply more to the documents representing that processed view. Let's say the ML determines that CostCo is normally "Office Supplies", printer paper and snacks for the office. It can see that your average CostCo bill is $300. But then you have a CostCo transaction that is $5,000. You purchased a laptop, phone and printer for the business. This shoebox item may have been processed to be categorized as "Office Supplies", however, because the total is so different than other transactions we flag the processing as "Draft/Needs Confirmation" and explain that because of the value of the transaction we think it may need closer attention. Perhaps the business owner classifies it as "Capital Asset" or something else to show it needs to be tracked and depreciated. Maybe they don't care and they leave it as "Office Supplies". After the user confirms then it really is processed.

This feels like something that we'll address later as a part of processing and perhaps just gives fuel to the argument that `processed` is deferred for now but I wanted to bring up the scenario in case it enables someone else to bring clarity to what we're modeling.